modules/contact/src/userobjects/PenaltyWeightedGapUserObject.C

```diff
@@ -68,25 +143,152 @@ PenaltyWeightedGapUserObject::getNormalContactPressure(const Node * const node)
     return 0.0;
 }
 
+Real
+PenaltyWeightedGapUserObject::getNormalLagrangeMultiplier(const Node * const node) const
+{
+  const auto it = _dof_to_lagrange_multiplier.find(_subproblem.mesh().nodePtr(node->id()));
+
+  if (it != _dof_to_lagrange_multiplier.end())
+    return -MetaPhysicL::raw_value(it->second);
+  else
+    return 0.0;
+}
+
 void
 PenaltyWeightedGapUserObject::reinit()
 {
   _contact_force.resize(_qrule_msm->n_points());
   for (const auto qp : make_range(_qrule_msm->n_points()))
-    _contact_force[qp] = 0;
+    _contact_force[qp] = 0.0;
 
   for (const auto i : make_range(_test->size()))
   {
     const Node * const node = _lower_secondary_elem->node_ptr(i);
-    const auto & weighted_gap =
-        libmesh_map_find(_dof_to_weighted_gap, static_cast<const DofObject *>(node)).first;
-    const auto weighted_gap_for_calc = weighted_gap < 0 ? -weighted_gap : ADReal(0);
-    const auto & test_i = (*_test)[i];
+
     for (const auto qp : make_range(_qrule_msm->n_points()))
+      _contact_force[qp] += (*_test)[i][qp] * _dof_to_normal_pressure[node];
+  }
+}
+
+void
+PenaltyWeightedGapUserObject::selfTimestepSetup()
+{
+  // clear the LMs
+  for (auto & dof_lm : _dof_to_lagrange_multiplier)
+    dof_lm.second = 0.0;
+
+  // reset penalty
+  for (auto & dof_lp : _dof_to_local_penalty)
+    dof_lp.second = _penalty;
+
+  // clear previous gap
+  for (auto & dof_pg : _dof_to_previous_gap)
+    dof_pg.second = 0.0;
+
+  // save old timestep
+  _dt_old = _dt;
+}
+
+void
+PenaltyWeightedGapUserObject::timestepSetup()
+{
+  selfTimestepSetup();
+}
+
+bool
+PenaltyWeightedGapUserObject::isAugmentedLagrangianConverged()
+{
+  Real positive_gap = 0.0;
```

Comment:
I think this would be more clear if these had 'max' or 'extreme' or something in their names to better convey what they are.